src/renderers/dom/client/validateDOMNesting.js

```diff
@@ -385,7 +398,17 @@ if (__DEV__) {
       didWarn[warnKey] = true;
 
       var tagDisplayName = childTag;
-      if (childTag !== '#text') {
+      var whitespaceInfo = '';
+      if (childTag === '#text') {
+        if (/\S/.test(childText)) {
+          tagDisplayName = 'Text nodes';
+        } else {
+          tagDisplayName = 'Whitespace text nodes';
+          whitespaceInfo =
+            ' Make sure you don\'t have any extra whitespace between tags on ' +
```

Comment:
Aha, didn't know that. I guess that makes sense, but it's not something you normally think about! (Which is why this warning is helpful!)